### ISSUE TITLE
[bug] conanbuildinfo.xcconfig is missing dependency frameworks

### ISSUE THREAD

**Issue by w3sip** [conan-io/conan#9878]
When I build a recipe depending (directly, but mostly transitively) on a number of other recipes which generate iOS framework artifacts, there's a difference in behavior between 1.23.0 and 1.41.0.

The difference seems to be in the generated conanbuildinfo.xcconfig. HEADER_SEARCH_PATHS and LIBRARY_SEARCH_PATHS seem to be similar enough to not be a problem. However, OTHER_LDFLAGS under 1.23.0 contains every framework dependency as `-framework MyFramework`. In 1.41.0 they're all gone, causing a failure to link.

conanbuildinfo.txt still contains all the relevant frameworks in the `[frameworks]` section in both versions.

**Comment by w3sip** [comment]
Looks like it is still operational in 1.25.2, and broken in 1.26.0 and on.

**Comment by w3sip** [comment]
The easiest way to reproduce:

```
from conans import ConanFile, tools
import os
from glob import glob
import shutil

class TestConan(ConanFile):
    generators = "xcode"
    name = "test"
    version = "1.0"

    def requirements(self):
        self.requires("dlib/19.21.34+01ab04dd@si/branch_release_19.21")


    def build(self):
        pass
```

With 1.25.2, after `conan install`
```
$ cat conanbuildinfo.xcconfig

HEADER_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/include"
LIBRARY_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/lib"
OTHER_LDFLAGS = $(inherited)    -framework dlib

GCC_PREPROCESSOR_DEFINITIONS = $(inherited)
OTHER_CFLAGS = $(inherited)
OTHER_CPLUSPLUSFLAGS = $(inherited)
FRAMEWORK_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da" /Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/lib
```


With 1.41.0, after `conan install`
```
$ cat conanbuildinfo.xcconfig

HEADER_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/include"
LIBRARY_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/lib"
OTHER_LDFLAGS = $(inherited)

GCC_PREPROCESSOR_DEFINITIONS = $(inherited)
OTHER_CFLAGS = $(inherited)
OTHER_CPLUSPLUSFLAGS = $(inherited)
FRAMEWORK_SEARCH_PATHS = $(inherited) "/Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da" /Users/alex/.conan/data/dlib/19.21.34+01ab04dd/si/branch_release_19.21/package/5f815ecc0dc3aa08b3ea3d5f4ac2836f436215da/lib
```

**Comment by w3sip** [comment]
This seems to be broken only if conanfile settings do not have `compiler`.  This is the change in `xcode.py` causing this:
```
-        self.frameworks = " ".join(format_frameworks(deps_cpp_info.frameworks, compiler="apple-clang"))
+        self.frameworks = " ".join(format_frameworks(deps_cpp_info.frameworks, self.conanfile.settings))

```

**Comment by memsharded** [comment]
Hi @w3sip 

We are actively working on a new generator ``XcodeDeps`` that will be released in next 1.42 very soon. I'd recommend to start using it when it is released, it will be more maintained and moved forward to 2.0.

Lets see if @SSE4 can have a look to this (if it is a possible and no-risk fix to legacy ``xcode`` generator)
